app/src/main/kotlin/com/wire/android/ui/home/conversations/model/MessageTypes.kt

```diff
@@ -0,0 +1,176 @@
+            )
+            .border(
+                width = 1.dp,
+                color = MaterialTheme.wireColorScheme.secondaryButtonDisabledOutline,
+                shape = RoundedCornerShape(MaterialTheme.wireDimensions.messageAssetBorderRadius)
+            )
+            .clickable { onAssetClick() }
+            .padding(MaterialTheme.wireDimensions.spacing8x)
+    ) {
+        Column {
+            Text(
+                text = assetName,
+                style = MaterialTheme.wireTypography.body02,
+                maxLines = 2,
+                overflow = TextOverflow.Ellipsis,
+            )
+            ConstraintLayout(Modifier.fillMaxWidth().padding(top = MaterialTheme.wireDimensions.spacing8x)) {
+                val (icon, description, downloadStatus) = createRefs()
+                Image(
+                    modifier = Modifier
+                        .constrainAs(icon) {
+                            top.linkTo(parent.top)
+                            start.linkTo(parent.start)
+                            bottom.linkTo(parent.bottom)
+                        },
+                    painter = painterResource(R.drawable.ic_file),
+                    contentDescription = stringResource(R.string.content_description_image_message),
+                    colorFilter = ColorFilter.tint(MaterialTheme.wireColorScheme.badge)
+                )
+                Text(
+                    modifier = Modifier
+                        .padding(start = MaterialTheme.wireDimensions.spacing8x)
+                        .constrainAs(description) {
+                            top.linkTo(parent.top)
+                            start.linkTo(icon.end)
+                            bottom.linkTo(parent.bottom)
+                        },
+                    text = assetDescription,
+                    style = MaterialTheme.wireTypography.subline01
+                )
+                Text(
+                    modifier = Modifier
+                        .padding(start = MaterialTheme.wireDimensions.spacing8x)
+                        .constrainAs(downloadStatus) {
+                            top.linkTo(parent.top)
+                            end.linkTo(parent.end)
+                            bottom.linkTo(parent.bottom)
+                        },
+                    text = stringResource(R.string.asset_message_download_text),
+                    style = MaterialTheme.wireTypography.subline01
+                )
+            }
+        }
+    }
+}
+
+private fun provideAssetDescription(assetExtension: String, assetSize: Int): String {
+    return when {
+        assetSize < 1000 -> "${assetExtension.uppercase()} ($assetSize B)"
+        assetSize in 1000..999999 -> "${assetExtension.uppercase()} (${assetSize / 1000} KB)"
+        else -> "${assetExtension.uppercase()} (${((assetSize / 1000000f) * 100.0).roundToInt() / 100.0} MB)" // 2 decimals round off
+    }
+}
```

Comment:
I think detekt is complaining about these magic numbers, maybe we could extract the numbers values to some constants 🤔  or just adding the annotation to avoid the check, should do it :D